contracts/Sacd.sol

```diff
@@ -96,12 +92,39 @@ contract Sacd is ISacd, Initializable, AccessControlUpgradeable, UUPSUpgradeable
         revert ZeroAddress();
       }
 
+      // Validate template permissions if template is used
       SacdStorage storage $ = _getSacdStorage();
 
-      uint256 tokenIdVersion = $.tokenIdToVersion[asset][tokenId];
-      $.permissionRecords[asset][tokenId][tokenIdVersion][grantee] = PermissionRecord(permissions, expiration, source);
+      if (templateId != 0) {
+        address templateContract = $.templateContract;
+
+        if (templateContract != address(0)) {
```

Comment:
If `templateContract` is not set and `templateId != 0` then I think we should revert.

Reply:
Fair enough, we can do that